src/main/routing/topology.c

```diff
@@ -1823,8 +1823,20 @@ static gboolean _topology_computeSourcePaths(Topology* top, igraph_integer_t src
     gboolean foundDstPosition = FALSE;
     GString* pathStringBuffer = g_string_new(NULL);
 
+    // So we ran the shortest path algorithm, which gave us a list of vertices that make
+    // up the path between the source and the destination (stored in resultPaths).
+    // We have the following:
+    // - If resultPaths contains a single entry, that entry stores the destination vertex.
+    // - Otherwise, the first entry stores the source vertex, and the destination vertex
+    //   is stored in the last entry.
+    // So now we go through and cache all of the incremental shortest paths that we
+    // happened to compute while building the full path between source and destination.
+
+    // Skip over the source if it is included in the resultPaths list.
+    gint startingPosition = numTargets == 1 ? 0 : 1;
+
     /* go through the result paths for all targets */
-    for(gint position = 0; position < numTargets; position++) {
+    for (gint position = startingPosition; position < numTargets; position++) {
```

Comment:
I don't really understand the comments here. My understanding is that `resultPaths` contains a list of lists of vertices (a path for each target vertex), but the comments only seem to be discussing a specific path.

> Otherwise, the first entry stores the source vertex, and the destination vertex is stored in the last entry.

Won't the `resultPaths` element where the target is the destination vertex be stored in entry `dstVertexIndexPosition`? And why is it assumed that the first `resultPaths` element is a path to the source vertex? Doesn't this depend on the output of `_topology_getUniqueVertexTargets` where the targets may be in any order?

Reply:
Yeah you're right! This change is incorrect and I'll need to fix it.